test/Grpc.Net.Client.Tests/CallCredentialTests.cs

```diff
@@ -158,6 +158,147 @@ public async Task CallCredentialsWithHttp_UnsafeUseInsecureChannelCallCredential
+        await call.ResponseAsync.DefaultTimeout();
+
+        // Assert
+        Assert.AreEqual("SECRET_TOKEN", authorizationValue);
+    }
+
+    [Test]
+    public void CallCredentialsWithHttp_ChannelCredentials_Error()
+    {
+        // Arrange
+        string? authorizationValue = null;
+        var httpClient = ClientTestHelpers.CreateTestClient(async request =>
+        {
+            authorizationValue = request.Headers.GetValues("authorization").Single();
+
+            var reply = new HelloReply { Message = "Hello world" };
+            var streamContent = await ClientTestHelpers.CreateResponseContent(reply).DefaultTimeout();
+            return ResponseUtils.CreateResponse(HttpStatusCode.OK, streamContent);
+        }, new Uri("http://localhost"));
+        var callCredentials = CallCredentials.FromInterceptor(async (context, metadata) =>
+        {
+            // The operation is asynchronous to ensure delegate is awaited
+            await Task.Delay(50);
+            metadata.Add("authorization", "SECRET_TOKEN");
+        });
+
+        // Act
+        var ex = Assert.Throws<InvalidOperationException>(() =>
+        {
+            _ = HttpClientCallInvokerFactory.Create(
+                httpClient,
+                configure: o =>
+                {
+                    o.Credentials = ChannelCredentials.Create(ChannelCredentials.Insecure, callCredentials);
+                });
+        })!;
+
+        // Assert
+        Assert.AreEqual("CallCredentials can't be composed with InsecureCredentials. " +
+            "CallCredentials must be used with secure channel credentials like SslCredentials " +
+            "or by enabling GrpcChannelOptions.UnsafeUseInsecureChannelCallCredentials on the channel.", ex.Message);
+    }
+
+    [TestCase(true, "https://localhost")]
+    [TestCase(false, "http://localhost")]
+    public void CallCredentialsWithHttp_CustomChannelCredentials_CheckSecureStatus(bool isSecure, string address)
+    {
+        // Arrange
+        string? authorizationValue = null;
+        var httpClient = ClientTestHelpers.CreateTestClient(async request =>
+        {
+            authorizationValue = request.Headers.GetValues("authorization").Single();
+
+            var reply = new HelloReply { Message = "Hello world" };
+            var streamContent = await ClientTestHelpers.CreateResponseContent(reply).DefaultTimeout();
+            return ResponseUtils.CreateResponse(HttpStatusCode.OK, streamContent);
+        }, new Uri(address));
+        var callCredentials = CallCredentials.FromInterceptor(async (context, metadata) =>
+        {
+            // The operation is asynchronous to ensure delegate is awaited
+            await Task.Delay(50);
+            metadata.Add("authorization", "SECRET_TOKEN");
+        });
+
+        if (isSecure)
+        {
+            // Act && Assert
+            CreateInvoker(httpClient, isSecure, callCredentials);
```

Comment:
Is it meaningful to add an assertion that uses the constructed invoker to issue a call and checks the token?